### PR TITLE
Filter logs from DB by AuthToken time period

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

There is a chance to send logs from different users in one batch. To avoid that we need to filter stored logs by token's valid time.

## Related PRs or issues

[AB#58523](https://msmobilecenter.visualstudio.com/web/wi.aspx?pcguid=27bf89de-9697-418a-accf-1aa125b22914&id=58523)

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorageVersion.h [hunk 1]
```diff
@@ -7,5 +7,6 @@ static NSUInteger const kMSInitialVersion = 0;
 static NSUInteger const kMSTargetTokenVersion = 1;
 static NSUInteger const kMSTargetKeyVersion = 2;
 static NSUInteger const kMSLogPersistencePriorityVersion = 3;
+static NSUInteger const kMSTimestmpVersion = 4;
```
Comment (annakocheshkova) [inline AppCenter/AppCenter/Internals/Storage/MSLogDBStorageVersion.h]:
```suggestion
static NSUInteger const kMSTimestampVersion = 4;
```

File: AppCenter/AppCenterTests/MSLogDBStorageTests.m [hunk 2]
```diff
@@ -941,20 +980,14 @@ - (void)testMigrationFromSchema1to3 {
...
-                              @"\"priority\" INTEGER)"));
+  assertThat(currentTable, is(kMSLatestSchema));
   NSString *priorityIndex =
       [self.sut executeSelectionQuery:[NSString stringWithFormat:@"SELECT sql FROM sqlite_master WHERE name='ix_%@_%@'", kMSLogTableName,
                                                                  kMSPriorityColumnName]][0][0];
   assertThat(priorityIndex, is(@"CREATE INDEX \"ix_logs_priority\" ON \"logs\" (\"priority\")"));
 }
 
-- (void)testMigrationFromSchema2to3 {
+- (void)testMigrationFromSchema2toLatest {
```
Comment (ElektrojungeAtWork) [inline AppCenter/AppCenterTests/MSLogDBStorageTests.m]: I like that we renamed the test methods for DB migration.

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m [hunk 3]
```diff
@@ -379,6 +387,11 @@ - (void)migrateDatabase:(void *)db fromVersion:(NSUInteger)version {
     [MSDBStorage executeNonSelectionQuery:migrationQuery inOpenedDatabase:db];
     [self createPriorityIndex:db];
```
Comment (guperrot) [inline AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: Don't we need an index for timestamp by the way? We're going to frequently query based on it?

Comment (MatkovIvan) [reply AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: Good topic to discuss. Let's do it separately

**Pull request merged**